src/Microsoft.IdentityModel.Tokens/AsymmetricAdapter.cs

```diff
@@ -53,36 +55,29 @@ internal class AsymmetricAdapter : IDisposable
         // which returns RSACng(). Our 4.5 target doesn't know about this type and sees it as RSA, then things start to go bad.
         // We use reflection to detect that 4.6+ is available and access the appropriate signing or verifying methods.
         private static Type _hashAlgorithmNameType = typeof(object).Assembly.GetType("System.Security.Cryptography.HashAlgorithmName", false);
+        private static Type _rsaEncryptionPaddingType = typeof(object).Assembly.GetType("System.Security.Cryptography.RSAEncryptionPadding", false);
         private static Type _rsaSignaturePaddingType = typeof(object).Assembly.GetType("System.Security.Cryptography.RSASignaturePadding", false);
-        private static volatile Func<RSA, byte[], string, byte[]> _rsaPkcs1SignMethod;
-        private static volatile Func<RSA, byte[], byte[], string, bool> _rsaPkcs1VerifyMethod;
+
+        private volatile Func<RSA, byte[], byte[]> _rsaDecrypt45Method;
+        private volatile Func<RSA, byte[], byte[]> _rsaEncrypt45Method;
+        private volatile Func<RSA, byte[], string, byte[]> _rsaPkcs1SignMethod;
```

Comment:
I see this is existing, but I'm not quite following why this needs to be volatile

Reply:
I don't think it is necessary since these are instance variables.
I will remove this, thanks.